sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -48,6 +49,21 @@ case class Mode(
 
   override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType)
 
+  override def checkInputDataTypes(): TypeCheckResult = {
+    val defaultCheck = super.checkInputDataTypes()
+    if (defaultCheck.isFailure) {
+      defaultCheck
+    } else if (UnsafeRowUtils.isBinaryStable(child.dataType) ||
+      child.dataType.isInstanceOf[StringType]) {
```

Comment:
Isn't this expression always going to be `True` for `StringType` regardless of collation? I don't see how we can end up in `else` branch?

Reply:
else branch covers complex types containing collated string, for example: `array<string collate unicode>`

Reply:
I see.
Please change error message then. `isBinaryStable` might evolve to include more types than just collations in future. It should be enough to just say that input of <type> is not supported by mode.

Reply:
Also, maybe this should be easier to read:
```scala
if (UnsafeRowUtils.isBinaryStable(child.dataType) || child.dataType.isInstanceOf[StringType]) {
  super.checkInputDataTypes()  
} else {
  TypeCheckResult.TypeCheckFailure(...)
}
```

Reply:
done!